sql: TestInternalExecAppNameInitialization failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/d6adf24cae788d7cd967feadae8e9c0388ce5273

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestInternalExecAppNameInitialization PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1080705&tab=buildLog

```
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1791 +0x392
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x14a

Goroutine 527 (finished) created at:
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:678 +0xb8
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:424 +0x1483
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:464 +0x87a
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:676 +0x16e
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func2()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:631 +0x53
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func3()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:639 +0x34
==================
I190104 08:01:50.851719 62766 sql/event_log.go:131  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.54.16.172:34185 Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181217-398-gd6adf24 StartedAt:1546588909761569692 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:a8cd05af-99cd-4d58-960e-2a4ac1035f15 StartedAt:1546588909761569692 LastUp:1546588909761569692}
I190104 08:01:50.890771 62546 util/stop/stopper.go:536  quiescing; tasks left:
2      node.Node: batch
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190104 08:01:50.891393 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190104 08:01:50.891874 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
I190104 08:01:50.892331 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
W190104 08:01:50.892567 62417 internal/client/txn.go:507  [n1,split,s1,r19/1:/{Table/23-Max}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190104 08:01:50.892842 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
1      [async] storage.split: processing replica
E190104 08:01:50.893222 62417 storage/queue.go:844  [n1,split,s1,r19/1:/{Table/23-Max}] unable to split [n1,s1,r19/1:/{Table/23-Max}] at key "/Table/24": split at key /Table/24 failed: result is ambiguous (server shutdown)
I190104 08:01:50.893650 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
I190104 08:01:50.909915 62508 kv/transport_race.go:113  transport race promotion: ran 19 iterations on up to 261 requests
    --- FAIL: TestInternalExecAppNameInitialization/session_bound_exec (1.31s)
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test

```

SHA: https://github.com/cockroachdb/cockroach/commits/99064bf2797cce49e2d56d0324a3344cf2312aca

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestInternalExecAppNameInitialization PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1082844&tab=buildLog

```
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:240 +0x141
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:224 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1793 +0x139
  github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:163 +0x142
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1791 +0x392
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x14a

Goroutine 622 (finished) created at:
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:678 +0xb8
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:424 +0x1483
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:464 +0x87a
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:676 +0x16e
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func2()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:631 +0x53
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func3()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:639 +0x34
==================
I190106 06:59:49.574805 62842 server/server_update.go:66  [n1] no need to upgrade, cluster already at the newest version
I190106 06:59:49.598285 62845 sql/event_log.go:131  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.208.41.114:38049 Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181217-435-g99064bf StartedAt:1546757988342450719 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:cd9ac0bc-2daf-4328-97b8-a7a53ace0b6b StartedAt:1546757988342450719 LastUp:1546757988342450719}
I190106 06:59:49.620965 62533 util/stop/stopper.go:536  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190106 06:59:49.623738 62533 util/stop/stopper.go:536  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] closedts-rangefeed-subscriber
W190106 06:59:49.624924 62707 storage/store.go:1647  [n1,s1,r19/1:/Table/2{3-4}] unable to gossip on capacity change: node unavailable; try another peer
I190106 06:59:49.626335 62533 util/stop/stopper.go:536  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] storage.split: processing replica
W190106 06:59:49.628667 62840 storage/replica.go:3279  [n1,s1,r19/1:/Table/2{3-4}] during async intent resolution: node unavailable; try another peer
I190106 06:59:49.630736 62533 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
I190106 06:59:49.673817 62495 kv/transport_race.go:113  transport race promotion: ran 24 iterations on up to 281 requests
    --- FAIL: TestInternalExecAppNameInitialization/session_bound_exec (1.49s)
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test

```

SHA: https://github.com/cockroachdb/cockroach/commits/dadcaa493bca02996ebd87c0c85106e566893ce2

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestInternalExecAppNameInitialization PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1084665&tab=buildLog

```
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:464 +0x87a
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:676 +0x16e
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func2()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:631 +0x53
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func3()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:639 +0x34

Goroutine 752 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:203 +0x596
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:490 +0x766
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:240 +0x141
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:224 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1793 +0x139
  github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:163 +0x142
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1791 +0x392
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x14a
==================
I190108 07:37:37.347609 63378 sql/event_log.go:135  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.210.129.120:34653 Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181217-468-gdadcaa4 StartedAt:1546933056133584604 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:be46e635-07a1-454b-b0e4-070b5c7de92b StartedAt:1546933056133584604 LastUp:1546933056133584604}
I190108 07:37:37.417057 63029 util/stop/stopper.go:536  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190108 07:37:37.417846 63029 util/stop/stopper.go:536  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] closedts-rangefeed-subscriber
W190108 07:37:37.421992 63210 storage/store.go:1646  [n1,s1,r19/1:/Table/2{3-4}] unable to gossip on capacity change: node unavailable; try another peer
W190108 07:37:37.425233 63087 storage/replica.go:3056  [n1,s1,r19/1:/Table/2{3-4}] during async intent resolution: node unavailable; try another peer
I190108 07:37:37.425884 63029 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] closedts-rangefeed-subscriber
I190108 07:37:37.426363 63029 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I190108 07:37:37.444422 63029 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
I190108 07:37:37.470214 63284 kv/transport_race.go:113  transport race promotion: ran 23 iterations on up to 292 requests
    --- FAIL: TestInternalExecAppNameInitialization/session_bound_exec (1.58s)
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test

```

--- a/pkg/sql/lease.go
+++ b/pkg/sql/lease.go
@@ -208,7 +208,7 @@ func (s LeaseStore) acquire(ctx context.Context, tableID sqlbase.ID) (*tableVers
 		// read from the database for the special descriptor of a system table
 		// (#23937).
 		insertLease := fmt.Sprintf(
-			`INSERT INTO system.lease ("descID", version, "nodeID", expiration) VALUES (%d, %d, %d, %s)`,
+			`INSERT INTO system.public.lease ("descID", version, "nodeID", expiration) VALUES (%d, %d, %d, %s)`,
 			storedLease.id, storedLease.version, nodeID, &storedLease.expiration,
 		)
 		count, err := s.execCfg.InternalExecutor.Exec(ctx, "lease-insert", txn, insertLease)
@@ -242,7 +242,7 @@ func (s LeaseStore) release(ctx context.Context, stopper *stop.Stopper, lease *s
 		if nodeID == 0 {
 			panic("zero nodeID")
 		}
-		const deleteLease = `DELETE FROM system.lease ` +
+		const deleteLease = `DELETE FROM system.public.lease ` +
 			`WHERE ("descID", version, "nodeID", expiration) = ($1, $2, $3, $4)`
 		count, err := s.execCfg.InternalExecutor.Exec(
 			ctx,
@@ -469,7 +469,7 @@ func CountLeases(
 		)
 	}
 
-	stmt := fmt.Sprintf(`SELECT count(1) FROM system.lease AS OF SYSTEM TIME %s WHERE `,
+	stmt := fmt.Sprintf(`SELECT count(1) FROM system.public.lease AS OF SYSTEM TIME %s WHERE `,
 		at.AsOfSystemTime()) +
 		strings.Join(whereClauses, " OR ")
 	values, err := executor.QueryRow(
@@ -1782,7 +1782,7 @@ func (m *LeaseManager) DeleteOrphanedLeases(timeThreshold int64) {
 
 		// Read orphaned leases.
 		sqlQuery := fmt.Sprintf(`
-SELECT "descID", version, expiration FROM system.lease AS OF SYSTEM TIME %d WHERE "nodeID" = %d
+SELECT "descID", version, expiration FROM system.public.lease AS OF SYSTEM TIME %d WHERE "nodeID" = %d
 `, timeThreshold, nodeID)
 		var rows []tree.Datums
 		retryOptions := base.DefaultRetryOptions()

